### PR TITLE
[BME-5] Added a check make sure setS3Auth is run prior to running Zep start

### PR DESCRIPTION
**Pull request by dustinvanstee**
Added a status flag to make sure setS3Auth is run before z.start.  If not, an error message will print.  Also added a simple check for making sure rootkey.csv exists.

added a new status flag s3authSet = False
once setS3Auth is run, then this flag is set True
start method checks this flag prior to start.   if not set, prints
error message and aborts ..

### CODE REVIEW

File: data/services/zeppelin.py
Comment (rawkintrevo) [inline data/services/zeppelin.py:56]: +1 on this, much more elegent

File: data/services/zeppelin.py
Comment (rawkintrevo) [inline data/services/zeppelin.py]: I think before it would have failed on a 'IOException File Not Found' for rootkey.csv not being present (though it might not have deadlined the program.  At anyrate- we should raise an exception here, not simply (and very quietly) print an error message on what will eventually be a catastrophic issue.

File: data/services/zeppelin.py
Comment (dustinvanstee) [reply data/services/zeppelin.py:56]: performed some simple manual tests ...
1.  tested is s3setAuth not run prior to zeppelin start
2.  tested with rootkey.csv file missing 
both tests passed as expected

**Review by rawkintrevo — DISMISSED** [review]

File: data/services/zeppelin.py
Comment (dustinvanstee) [reply data/services/zeppelin.py]: +1  I will add that and re-issue pr

Comment (rawkintrevo) [reply data/services/zeppelin.py]: if you just recommit the pr will update automatically

File: data/services/zeppelin.py
Comment (rawkintrevo) [inline data/services/zeppelin.py:39]: This introduces API breaking changes.  The example scripts need to be updated from:

```python
z = ZeppelinServiceOnBI(SERVER, USERNAME, PASSWORD)
z.install()
z.start()
sleep(5)
z.downloadConfig({"interpreter.json"  : "conf/interpreter.json"})
z.updateConfig()
#z.setS3auth(S3_USERNAME, S3_BUCKET) ## Must do this first!
```

to 

```python
z = ZeppelinServiceOnBI(SERVER, USERNAME, PASSWORD)
z.install()
z.setS3auth(S3_USERNAME, S3_BUCKET) ## Must do this first!
z.start()
sleep(5)
z.downloadConfig({"interpreter.json"  : "conf/interpreter.json"})
z.updateConfig()
```

E.g. moving `setS3auth(...)` to before the first `z.start()`

File: data/services/zeppelin.py
Comment (rawkintrevo) [inline data/services/zeppelin.py]: add `or specify alternate path`

File: data/services/zeppelin.py
Comment (rawkintrevo) [inline data/services/zeppelin.py]: `exit(1)` to indicate failure.

**Comment by rawkintrevo** [comment]
Looking good @dustinvanstee . A couple of more comments to address after testing then should be good to merge. 

Thanks!

**Comment by dustinvanstee** [comment]
Addressed comments above, and performed some tests using the example deployment scripts ... Looks good to me.  Code re-committed

File: example_zeppelin_mahout.py
Comment (rawkintrevo) [inline example_zeppelin_mahout.py:51]: Need to clean up the merge

**Comment by rawkintrevo** [comment]
lgtm if no further comment

**Pull request merged**
